src/storage/ExitChannelBlockStorage.ts

```diff
@@ -0,0 +1,82 @@
+import { ethers } from "ethers";
+import {
+    ExitChannelBlockStruct,
+    BalanceStruct
+} from "@typechain-types/contracts/V1/types/DataTypes";
+import { Hash } from "@/types/types";
+import { Codec, Type } from "@/utils";
+
+export class ExitChannelBlockStorage {
+    private blockMap: Map<Hash, ExitChannelBlockStruct>;
+    private _latestBlockHash: Hash;
+    private _totalWithdrawals: BalanceStruct;
+
+    constructor() {
+        this.blockMap = new Map();
+        this._latestBlockHash = ethers.ZeroHash;
+        this._totalWithdrawals = {
+            amount: BigInt(0),
+            data: "0x"
+        };
+    }
+
+    // ====================================
+    // CREATE
+    // ====================================
+
+    // Exit Channel Block
+
+    /** [OVERLOAD 1] Store exit channel block with auto-computed hash */
+    storeExitChannelBlock(block: ExitChannelBlockStruct): Hash;
+
+    /** [OVERLOAD 2] Store exit channel block with provided hash */
+    storeExitChannelBlock(block: ExitChannelBlockStruct, blockHash: Hash): Hash;
+
+    storeExitChannelBlock(
+        block: ExitChannelBlockStruct,
+        blockHash?: Hash
+    ): Hash {
+        const hash =
+            blockHash ??
+            ethers.keccak256(Codec.encode(block, Type.ExitChannelBlock));
+
+        // Check for duplicates
+        if (this.blockMap.has(hash)) {
+            throw new Error(
+                `Exit channel block with hash ${hash} already exists`
```

Comment:
No need to error here - if it exists just return the same hash - it's possible to trigger this for the same `exitChannelBlock` multiple times if a reorg/fork happens - probably same applies for `JoinChannelBlock`